### PR TITLE
Feature/appcli 106 configure and apply on install by default

### PR DESCRIPTION
**Pull request by benjamin-wiffen**

### CODE REVIEW

File: appcli/templates/installer.j2 [hunk 1]
```diff
@@ -31,6 +53,12 @@ function main()
     rm -f "${_launcher_generic}" || print_error_and_exit "No permissions to delete to [${_launcher_generic}]"
     ln -s "${_launcher_versioned}" "${_launcher_generic}"
 
+    # Auto configure if enabled.
+    if [ $AUTO_CONFIGURE = true ]; then
+        {{ install_dir }}/{{ cli_context.app_name|lower }} configure init
```
Comment (thomas-anderson-bsl) [inline appcli/templates/installer.j2:58]: Should reference the already-existing `${_launcher_generic}`

**Review by thomas-anderson-bsl — CHANGES_REQUESTED** [review]

File: appcli/templates/installer.j2 (hunk 1 above)
Comment (al-jeyapal) [reply appcli/templates/installer.j2:58]: Also use double square brackets for tests `[[`, its a better alternative to single square brackets `[`

**Review by al-jeyapal — CHANGES_REQUESTED** [review]

File: appcli/templates/installer.j2 [hunk 2]
```diff
@@ -85,8 +97,27 @@ function print_error_and_exit()
     exit 1
 }
 
+function args()
+{
+    options=$(getopt --long auto-configure --long no-auto-configure -- "$@")
```
Comment (al-jeyapal) [inline appcli/templates/installer.j2:102]: `getopt` is not a built in command, it's an external command which is not POSIX compliant.
It behaves differently on various Linux distros (e.g. on `Arch` the `--long` option is called `--longoptions`.
So this isn't very portable which is problematic as we need the script to be as portable as possible.
I'll update to reflect how we parse long args in our plummet example script.

File: appcli/templates/installer.j2 [hunk 3]
```diff
@@ -85,8 +97,27 @@ function print_error_and_exit()
     exit 1
 }
 
+function args()
```
Comment (al-jeyapal) [inline appcli/templates/installer.j2:100]: Method needs doco. Will update.l

File: appcli/templates/installer.j2 [hunk 4]
```diff
@@ -85,8 +97,27 @@ function print_error_and_exit()
...
+        shift
+    done
+}
+
 # ------------------------------------------------------------------------------
 # LOGIC
 # ------------------------------------------------------------------------------
 
-main $@
+args $0 "$@"
```
Comment (al-jeyapal) [inline appcli/templates/installer.j2:122]: It's not obvious why we call it like this, warrants a comment. Also, it should just be called from within `main` will update.

**Review by al-jeyapal — APPROVED** [review]

**Pull request merged**